Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -183,46 +183,44 @@ auto WebProcessProxy::globalPageMap() -> WebPageProxyMap&
     return pageMap;
 }
 
-Vector<RefPtr<WebPageProxy>> WebProcessProxy::globalPages()
+Vector<Ref<WebPageProxy>> WebProcessProxy::globalPages()
 {
-    return WTF::map(globalPageMap(), [] (auto& keyValue) -> RefPtr<WebPageProxy> {
+    return WTF::map(globalPageMap(), [] (auto& keyValue) -> Ref<WebPageProxy> {
         return keyValue.value.get();
     });
 }
 
-Vector<RefPtr<WebPageProxy>> WebProcessProxy::pages() const
+Vector<Ref<WebPageProxy>> WebProcessProxy::pages() const
 {
-    return WTF::map(m_pageMap, [] (auto& keyValue) -> RefPtr<WebPageProxy> {
+    return WTF::map(m_pageMap, [] (auto& keyValue) -> Ref<WebPageProxy> {
         return keyValue.value.get();
     });
 }
 
 void WebProcessProxy::forWebPagesWithOrigin(PAL::SessionID sessionID, const SecurityOriginData& origin, const Function<void(WebPageProxy&)>& callback)
 {
     for (auto& page : globalPages()) {
-        if (!page || page->sessionID() != sessionID || SecurityOriginData::fromURLWithoutStrictOpaqueness(URL { page->currentURL() }) != origin)
+        if (page->sessionID() != sessionID || SecurityOriginData::fromURLWithoutStrictOpaqueness(URL { page->currentURL() }) != origin)
             continue;
-        callback(*page);
+        callback(page);
     }
 }
 
 Vector<std::pair<WebCore::ProcessIdentifier, WebCore::RegistrableDomain>> WebProcessProxy::allowedFirstPartiesForCookies()
 {
     Vector<std::pair<WebCore::ProcessIdentifier, WebCore::RegistrableDomain>> result;
-    for (auto& page : globalPages()) {
-        if (page)
-            result.append(std::make_pair(page->process().coreProcessIdentifier(), RegistrableDomain(URL(page->currentURL()))));
-    }
+    for (auto& page : globalPages())
```

Comment:
See CheckedRef?

Reply:
auto& is already a `Ref&` here. globalPages() returns a `Vector<Ref<WebPageProxy>>`